pa4/checker-pa4.ml

```diff
@@ -30,23 +30,34 @@ let rec is_subtype t1 t2 =    (*checking if t1 is subtype of t2*)
 		with
 		| _ -> false (* x is undefined- Int, String, Bool*)
 		end
-	| _, _ -> (*TODO: check the class note like for SELF_TYPE*) false   
+	| SELF_TYPE(x), SELF_TYPE(y) -> 
+		if x = y then true
+		else false
+	| SELF_TYPE(c), Class(x) -> is_subtype (Class c) (Class x)
+	| Class(x), SELF_TYPE(c) -> false
+	| _, _ -> false   
 
 exception LowerBoundFound of string;;
 
-let rec lowest_upper_bound t1 t2 =
-	if (is_subtype t1 t2) then 
-		t2 
-	else if (is_subtype t2 t1) then 
-		t1 
+let rec lowest_upper_bound type_1 type_2 =
+	let real_type_1 = match type_1 with 
+	| SELF_TYPE(c) -> Class c
+	| Class(c) -> type_1
+in
+	let real_type_2 = match type_2 with 
+	| SELF_TYPE(c) -> Class c 
+	| Class(c) -> type_2
+in 
+	if (is_subtype real_type_1 real_type_2) then real_type_2 
+	else if (is_subtype real_type_2 real_type_1) then real_type_1
 	else begin
 		try
-			let x = type_to_str t1 in
-			(* get parent maps of t1 and t2 *)
+			let x = type_to_str real_type_1 in
+			(* get parent maps of type_1 and type_2 *)
```

Comment:
Is this comment outdated? It looks like we only get a parent map for type 1, not type 2.